src/lib/viewers/media/DashViewer.js

```diff
@@ -741,11 +753,21 @@ class DashViewer extends VideoBaseViewer {
     loadUI() {
         super.loadUI();
 
-        if (this.hdVideoId !== -1) {
+        if (this.hasHD()) {
             this.mediaControls.enableHDSettings();
         }
     }
 
+    /**
+     * @inheritdoc
+     */
+    loadUIReact() {
+        super.loadUIReact();
+
+        this.selectedQuality = this.cache.get('media-quality') || 'auto';
+        this.handleQuality();
```

Comment:
I think we should try to avoid calling handlers outside of user-initiated events. What is the goal of this logic?

Reply:
I was trying to leverage `handleQuality()` which would update the player based on the quality value but I'll just duplicate that logic into `setQuality` since eventually `handleQuality` will go away